src/convert.rs

```diff
@@ -0,0 +1,238 @@
+use num::Float;
+
+use {Alpha, Rgb, Luma, Xyz, Yxy, Lab, Lch, Hsv, Hsl, Color};
+
+///FromColor provides conversion between the colors. It requires from_xyz and derives conversion
+///to other colors as a default from this. These defaults must be overridden when direct
+///conversion exists between colors. For example, Luma has direct conversion to Rgb.
+///So from_rgb conversion for Luma and from_luma for Rgb is implemented directly.
+pub trait FromColor<T>: Sized
+    where T: Float,
+{
+    //To implement ColorConvert for a type, a few things must by kept in mind.
+    //
+    //First, derived colors like Hsl must override from_rgb and convert from_xyz via Rgb.
+    //
+    //Second, the from for the same color must override the default. For example,
+    //from_rgb for Rgb will convert via Xyz which needs to be overridden with self to
+    //avoid the unnecessary converison.
+    //
+    //Third, if a direct transform exists between the colors, then it should be
+    //overridden. For example, Luma has direct conversion to Rgb. So from_rgb conversion
+    //for Luma and from_luma for Rgb is implemented directly.
```

Comment:
These comments are a bit redundant now, when the same info can be found in the documentation comments.